src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,204 @@
+        try {
+            final String[] desTime = parseDeadline(args);
+            return new DeadlineCommand(desTime[0], desTime[1]);
+        } catch (DukeException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private Command prepareEvent(String args) {
+        try {
+            final String[] desTime = parseEvent(args);
+            return new EventCommand(desTime[0], desTime[1]);
+        } catch (DukeException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private Command prepareFind(String args) {
+        final Matcher matcher = FIND_FORMAT.matcher(args.trim());
+        if (!matcher.matches()) {
+            return new InvalidCommand(Messages.MESSAGE_INVALID_ARGUMENTS);
+        }
+        return new FindCommand(args.trim());
+    }
+
+    private Command prepareColour(String args) {
+        final Matcher matcher = COLOUR_FORMAT.matcher(args.trim());
+        if (!matcher.matches()) {
+            return new InvalidCommand(Messages.MESSAGE_INVALID_ARGUMENTS);
+        }
+        return new ColourCommand(args.trim());
+    }
+
+    private Command prepareDelete(String args) {
+        try {
+            final int targetIndex = parseArgsAsDisplayedIndex(args);
+            return new DeleteCommand(targetIndex);
+        } catch (DukeException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private Command prepareMark(String args) {
+        try {
+            final int targetIndex = parseArgsAsDisplayedIndex(args);
+            return new MarkCommand(targetIndex);
+        } catch (DukeException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private Command prepareUnmark(String args) {
+        try {
+            final int targetIndex = parseArgsAsDisplayedIndex(args);
+            return new UnmarkCommand(targetIndex);
+        } catch (DukeException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private String parseTodo(String args) throws DukeException {
+        final Matcher matcher = TODO_FORMAT.matcher(args.trim());
+        if (!matcher.matches()) {
+            throw new DukeException(Messages.MESSAGE_EMPTY_DESCRIPTION);
+        }
+        return args.trim();
+    }
+
+    private String[] parseDeadline(String args) throws DukeException {
+        if (!args.contains("/by")) {
+            throw new DukeException(String.format(Messages.MESSAGE_MISSING_SEPARATOR, "/by"));
+        }
+        return parseArgsAsDescriptionAndDate(DEADLINE_FORMAT, args);
+    }
+
+    private String[] parseEvent(String args) throws DukeException {
+        if (!args.contains("/at")) {
+            throw new DukeException(String.format(Messages.MESSAGE_MISSING_SEPARATOR, "/at"));
+        }
+        return parseArgsAsDescriptionAndDate(EVENT_FORMAT, args);
+    }
+
+    private int parseArgsAsDisplayedIndex(String args) throws DukeException {
```

Comment:
Could we perhaps add JavaDocs for the methods so that we can make things a little bit clearer?